lib/main.dart

```diff
@@ -44,33 +44,67 @@ void main() async {
 class MyApp extends StatelessWidget {
   const MyApp({super.key});
 
-  // This widget is the root of your application.
   @override
   Widget build(BuildContext context) {
     return MultiBlocProvider(
       providers: <SingleChildWidget>[
         BlocProvider<ThemeSwitchBloc>(
           create: (_) => getIt<ThemeSwitchBloc>(),
         ),
+        BlocProvider<NetworkCubit>(
+          create: (_) => getIt<NetworkCubit>(),
+        ),
       ],
-      child: BlocBuilder<ThemeSwitchBloc, ThemeSwitchState>(
-        builder: (BuildContext context, ThemeSwitchState state) {
-          return ScreenUtilInit(
-            designSize: const Size(360, 690),
-            minTextAdapt: true,
-            splitScreenMode: true,
-            child: MaterialApp.router(
-              routerConfig: router,
-              title: AppStrings.appName,
-              builder: DevicePreview.appBuilder,
-              theme: AppTheme.lightTheme,
-              darkTheme: AppTheme.darkTheme,
-              themeMode: state.switchValue ? ThemeMode.dark : ThemeMode.light,
-              debugShowCheckedModeBanner: false,
-            ),
-          );
-        },
-      ),
+      child: const HomeScreen(),
+    );
+  }
+}
+
+class HomeScreen extends StatefulWidget {
```

Comment:
Why this code in home screen bro?

Reply:
@saigopin we need dispose method, when ever the app reloads.